index/scorch/introducer.go

```diff
@@ -110,6 +110,7 @@ func (s *Scorch) introduceSegment(next *segmentIntroduction) error {
 
 	// iterate through current segments
 	var running uint64
+	var index int
```

Comment:
one last thought, i think if we initialized segments and offsets with size 0 and capacity nsegs+1,
then we could simply use append, instead of need a new variable 'index'
also, this then removes the block of code at the end which tries to trim down the size of the slice

Reply:
thanks @mschoch , this is a new go learning for me that size 0 helps in unnecessary allocation towards reaching capacity when we know the exact max capacity!